### PR TITLE
SLING-11971 Adding additional logging for sling post processors

### PR DESCRIPTION
**Pull request by pat-lego**

### CODE REVIEW

**Comment by joerghoh** [comment]
@pat-lego Can you rebase to the latest master update? I fixed the Java17 build

File: src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java [hunk 1]
```diff
@@ -108,11 +112,15 @@ public void run(final SlingHttpServletRequest request,
             // invoke processors
             if (processors != null) {
                 for (SlingPostProcessor processor : processors) {
+                    Instant start = Instant.now();
+                    log.debug("About to perform processor {}", processor.getClass().getName());
                     processor.process(request, changes);
+                    log.debug("Performed processor {} in {}", processor.getClass().getName(), Duration.between(start, Instant.now()).toMillis());
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java]: Can you please also log this message to the RequestProgressTracker?

File: src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java [hunk 2]
```diff
@@ -106,7 +108,10 @@ public void run(final SlingHttpServletRequest request,
             try {
                 if (processors != null) {
                     for (SlingPostProcessor processor : processors) {
+                        Instant start = Instant.now();
+                        log.debug("About to perform processor {}", processor.getClass().getName());
                         processor.process(request, changes);
+                        log.debug("Performed processor {} in {}", processor.getClass().getName(), Duration.between(start, Instant.now()).toMillis());
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java:114]: can you please also log into the RequestProgressTracker?

**Review by joerghoh — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/sling/servlets/post/impl/operations/ModifyOperation.java [hunk 3]
```diff
@@ -70,31 +76,54 @@ public void setDateParser(final DateParser dateParser) {
...
         final Map<String, RequestProperty> reqProperties = collectContent(request, response);
 
         final VersioningConfiguration versioningConfiguration = getVersioningConfiguration(request);
 
         // do not change order unless you have a very good reason.
+        Instant start = Instant.now();
 
         // ensure root of new content
         processCreate(request.getResourceResolver(), reqProperties, response, changes, versioningConfiguration);
+        logger.debug("Performed create at {} path took {} ms to operate", request.getRequestURI(),
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/servlets/post/impl/operations/ModifyOperation.java:91]: I don't think that it is necessary to consistently add the value of ``request.getRequestURI()`` to the message. This information is always part of the context of the RequestProgressTracker, and also it won't change while execution it, it's static.
On top of that, ``Performed create at``  could be understood that a create is executed at this path, and this is not consistently true, as other resources could be affected, too (for example sub-resources).

I would just remove this part of the log.

File: src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java [hunk 4]
```diff
@@ -127,6 +135,7 @@ public void run(final SlingHttpServletRequest request,
             }
 
             // fail if any of the base paths (before the postfix) which had a postfix are contained in the modification set
+            log.debug("Fail if any of the base paths (before the postfix) which had a postfix are contained in the modification set");
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java:138]: What is the purpose of this statement? Is it just for the purpose of having a timestamp (assuming the log is properly configured to write mili/micro-second level timestamps)?

File: src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java [hunk 5]
```diff
@@ -167,13 +176,15 @@ public void run(final SlingHttpServletRequest request,
 
             if (isSessionSaveRequired(session, request)) {
                 request.getResourceResolver().commit();
+                log.debug("Saved the session");
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java:179]: same as above

File: src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java [hunk 6]
```diff
@@ -167,13 +176,15 @@ public void run(final SlingHttpServletRequest request,
...
             }
 
             if (!isSkipCheckin(request)) {
                 // now do the checkins
                 for(String checkinPath : nodesToCheckin) {
                     if (checkin(request.getResourceResolver(), checkinPath)) {
                         response.onChange("checkin", checkinPath);
+                        log.debug("Checked in the node");
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java:187]: same as above

**Review by joerghoh — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java [hunk 7]
```diff
@@ -62,7 +66,7 @@ public abstract class AbstractPostOperation implements PostOperation {
     /**
      * default log
      */
-    protected final Logger log = LoggerFactory.getLogger(getClass());
+    protected final Logger log = LoggerFactory.getLogger(AbstractPostOperation.class);
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java:69]: Why this change? now all subclasses log with the same logger, instead of using separate ones

File: src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java [hunk 8]
```diff
@@ -59,7 +61,7 @@ public abstract class AbstractPostOperation implements PostOperation {
     /**
      * Default logger
      */
-    protected final Logger log = LoggerFactory.getLogger(getClass());
+    protected final Logger log = LoggerFactory.getLogger(AbstractPostOperation.class);
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java:64]: Again, why change the logger?

File: src/main/java/org/apache/sling/servlets/post/impl/operations/ModifyOperation.java [hunk 9]
```diff
@@ -48,6 +52,8 @@
  */
 public class ModifyOperation extends AbstractCreateOperation {
 
+    Logger logger = LoggerFactory.getLogger(this.getClass());
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/servlets/post/impl/operations/ModifyOperation.java:55]: Why introducing a logger here? It gets inherited

File: src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java [hunk 10]
```diff
@@ -108,11 +112,15 @@ public void run(final SlingHttpServletRequest request,
             // invoke processors
             if (processors != null) {
                 for (SlingPostProcessor processor : processors) {
+                    Instant start = Instant.now();
+                    request.getRequestProgressTracker().log("About to perform processor {}", processor.getClass().getName());
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/servlets/post/AbstractPostOperation.java:116]: Why not use the timer function of the tracker?

File: src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java [hunk 11]
```diff
@@ -106,7 +108,10 @@ public void run(final SlingHttpServletRequest request,
             try {
                 if (processors != null) {
                     for (SlingPostProcessor processor : processors) {
+                        Instant start = Instant.now();
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java:111]: Same here, tracker provides already a timer

File: src/main/java/org/apache/sling/servlets/post/impl/operations/ModifyOperation.java [hunk 12]
```diff
@@ -134,13 +163,13 @@ protected String getResourcePath(SlingHttpServletRequest request) {
             // check whether it is a create request (trailing /)
             if (suffix.endsWith(SlingPostConstants.DEFAULT_CREATE_SUFFIX)) {
                 suffix = suffix.substring(0, suffix.length()
-                    - SlingPostConstants.DEFAULT_CREATE_SUFFIX.length());
+                        - SlingPostConstants.DEFAULT_CREATE_SUFFIX.length());
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/servlets/post/impl/operations/ModifyOperation.java:166]: From here on there are a lot of formatting changes; it would be good to leave them out of the PR

File: src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java (hunk 8 above)
Comment (henrykuijpers) [reply src/main/java/org/apache/sling/servlets/post/impl/operations/AbstractPostOperation.java:64]: This change is actually not desired, I would say, because now the log entries will always mention "AbstractPostOperation", instead of the inheritor?
